libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/permission/LocationPermissionComponent.kt

```diff
@@ -83,14 +83,8 @@ internal class LocationPermissionComponent(
     private fun notifyGrantedOnForegrounded(applicationContext: Context) {
         coroutineScope.launch {
             componentActivityRef.get()?.repeatOnLifecycle(Lifecycle.State.RESUMED) {
-                if (!store.state.value.tripSession.isLocationPermissionGranted) {
-                    val isGranted = PermissionsManager.areLocationPermissionsGranted(
-                        applicationContext
-                    )
-                    if (isGranted) {
-                        store.dispatch(TripSessionStarterAction.OnLocationPermission(true))
-                    }
-                }
+                val granted = PermissionsManager.areLocationPermissionsGranted(applicationContext)
+                store.dispatch(TripSessionStarterAction.OnLocationPermission(granted))
```

Comment:
yeah am updating it to stay in sync one way or another. for example, if you background the app to disable location permissions - the TripSessionStarter state should be updated

Reply:
kind of annoying there is not an android api for observing permissions. and we do not have one so, we could build one. Also related to this thread https://github.com/mapbox/mapbox-navigation-android/pull/6794#discussion_r1066108918

Reply:
New solution in place https://github.com/mapbox/mapbox-navigation-android/pull/6794#discussion_r1067562324. Resolving here

Reply:
> kind of annoying there is not an android api for observing permissions

I thinks it's because when you change the permission state in settings, you leave the app so you're expected to change permissions state in `onResume`.
You can also request the permission by yourself but in this case you kinda also receive the notification of the result.